azurelinuxagent/ga/exthandlers.py

```diff
@@ -1023,19 +1022,14 @@ def update(self, version=None, disable_exit_code=None, updating_from_version=Non
         env = {'VERSION': version, ExtCommandEnvVariable.DisableReturnCode: disable_exit_code,
                ExtCommandEnvVariable.UpdatingFromVersion: updating_from_version}
 
-        try:
-            self.set_operation(WALAEventOperation.Update)
-            man = self.load_manifest()
-            update_cmd = man.get_update_command()
-            self.logger.info("Update extension [{0}]".format(update_cmd))
-            self.launch_command(update_cmd,
-                                timeout=900,
-                                extension_error_code=ExtensionErrorCodes.PluginUpdateProcessingFailed,
-                                env=env)
-        except ExtensionError:
-            # prevent the handler update from being retried
-            self.set_handler_state(ExtHandlerState.Failed)
-            raise
+        self.set_operation(WALAEventOperation.Update)
+        man = self.load_manifest()
+        update_cmd = man.get_update_command()
+        self.logger.info("Update extension [{0}]".format(update_cmd))
+        self.launch_command(update_cmd,
+                            timeout=900,
+                            extension_error_code=ExtensionErrorCodes.PluginUpdateProcessingFailed,
+                            env=env)
```

Comment:
let's double-check report_ext_handler_status --- seems like the update failure would have been reported to crp

Reply:
Also double check the PR where they added the FAILED state, they did it to deal with going in the infinite loop I think. The PR would give you the motivation behind adding that and to see if this doesnt reintroduce the problem